wallet/wallet-rpc-lib/src/service/handle.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Handle used to control the wallet service
+
+use futures::future::{BoxFuture, Future};
+
+use utils::shallow_clone::ShallowClone;
+
+use super::worker::{self, WalletCommand, WalletController};
+
+/// Wallet handle allows the user to control the wallet service, perform queries etc.
+#[derive(Clone)]
+pub struct WalletHandle(worker::CommandSender);
```

Comment:
Yea. It's annoyingly close to subsystems (the RPC already is one out of the box). However the wallet service deals with the event loop differently from the other subsystems. It could be made into one by re-introducing custom event loops which I am not too keen on.

Alternatively, I could have a look how the `Handle` part of subsystems could be reused without pulling in the rest but then we only reuse the handle/call interface. We would not get the other things the subsystem crate implements like automatic clean shutdown and read-only calls parallelism. Especially the latter seems incompatible the structure of the wallet controller event loop. Even then some changes to the `subsytem` crate would be necessary even for this.